#! /bin/sh /usr/share/dpatch/dpatch-run
## 01cryptostyle-acinclude.dpatch by Julien Delange <julien@gunnm.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: fails to identify crypto++ release (#348579)

@DPATCH@
--- ../acinclude.m4.old	2006-03-26 08:14:12.000000000 +0200
+++ ../acinclude.m4	2006-03-26 08:15:47.000000000 +0200
@@ -586,8 +586,12 @@
 	  fi
 	  
 	  if test -f $crypto_prefix/include/crypto++/cryptlib.h; then
-		  CRYPTO_PP_STYLE="gentoo_debian"
-		  crypto_version=`grep "Reference Manual" $crypto_prefix/include/crypto++/cryptlib.h | sed -e's#.*\s\(\([0-9]\+\.\?\)\+\)\s.*#\1#g'`
+		if test -f /etc/debian_version; then
+			CRYPTO_PP_STYLE="debian"
+		else
+			CRYPTO_PP_STYLE="gentoo"
+		 fi
+		crypto_version=`grep "Reference Manual" $crypto_prefix/include/crypto++/cryptlib.h | sed 's/[[^0-9]]*\([[0-9]][[0-9.]]*\).*$/\1/'`
 	  fi
 
 	  vers=`echo $crypto_version | $AWK 'BEGIN { FS = "."; } { printf "% d", ([$]1 * 1000 + [$]2) * 1000 + [$]3;}'`
